include/libopencm3/stm32/h7/rcc.h

```diff
@@ -200,6 +439,37 @@ enum rcc_osc {
   RCC_LSI
 };
 
+enum rcc_sysclk_mux {
+  RCC_SYSCLK_PLL,
+  RCC_SYSCLK_HSE,
+  RCC_SYSCLK_HSI,
+};
+
+enum rcc_pll_mux {
+  RCC_PLL_HSI = RCC_PLLCKSELR_PLLSRC_HSI,
+  RCC_PLL_HSE = RCC_PLLCKSELR_PLLSRC_HSE
+};
+
+/** PLL Configuration structure. */
+struct rcc_pll_config {
+  uint32_t hse_frequency;           /**< User configured external crystal frequency. */
+  enum rcc_sysclk_mux sysclk_mux;   /**< SYSCLK source input selection. */
```

Comment:
Why not just the enum rcc_osc like others have?  G0 is probably the closest here.  I know h7 is the biggest and most complicated, but there's no reason to make it all up completely new and different.  You can just have the pll source as well, without the renaming enum on line 448.